### ISSUE TITLE
C++ indexing crash

### ISSUE THREAD

**Issue by egraether** [CoatiSoftware/CoatiBugTracker#318]
```
struct Lorem : public Ipsum<std::pair<int, int>>
{
  Lorem();
};

Lorem::Lorem()
{
  dolor([](Sit amet) -> Sit
    {
      return amet;
    });
}
```

log:
```
15:43:54 | 0x7000003a6000 | INFO: send MessageStatus Reindexing Project - loading
15:43:54 | 0x7000003a6000 | StatusBarController.cpp:62 setStatus() | INFO: STATUS Reindexing Project
15:43:54 | 0x7000003a6000 | INFO: send MessageClearErrorCount
15:43:54 | 0x7000003a6000 | INFO: send MessageWindowFocus
15:43:54 | 0x700000081000 | INFO: Indexer - TranslationUnitDecl <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | TypedefDecl [_..t] <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | BuiltinTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | TypedefDecl [_..t] <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | BuiltinTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | TypedefDecl [_..t] <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | PointerTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | | BuiltinTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | TypedefDecl [_..t] <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | ConstantArrayTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - | | | RecordTypeLoc <0:0, 0:0>
15:43:54 | 0x700000081000 | INFO: Indexer - Traversing "/Users/name/crash_coati/crash_coati.cpp"
15:43:54 | 0x700000081000 | INFO: Indexer - | CXXRecordDecl [L..m] <1:1, 1:8>
15:43:54 | 0x700000081000 | INFO: Indexer - | CXXRecordDecl [] <8:9, 8:9>
```

**Comment by mlangkabel** [comment]
reproducible on windows

**Comment by mlangkabel** [comment]
This crash is triggered by an assertion in the clang codebase. Throwing this probably didn't matter much to the clang devs because the snippet mentioned above wouldn't compile anyway.

We fixed the crash in Coati version 0.10.33 and just record an `undefined-lambda` in this case.
